src/OrbitSshQt/Session.cpp

```diff
@@ -132,7 +132,9 @@ Session::DisconnectResult Session::Disconnect() {
   if (CurrentState() == State::kConnected) {
     SetState(State::kAboutToDisconnect);
     OnEvent();
-    return DisconnectResult::kDisconnectStarted;
+    if (IsStopping()) {
+      return DisconnectResult::kDisconnectStarted;
```

Comment:
This is a bug I found while adding the tests. It can happen that the entire disconnect happens during this one `OnEvent` call in line 134. In that case we have to return `kDisconnectSuccessfully` is already done.

That's why we check if we are still in the process of stopping (`IsStopping`) after the `OnEvent` call and only return `kDisconnectStarted` in that case.